lib/davinci_dtr_test_kit/docs/dtr_full_ehr_suite_description_v201.md

```diff
@@ -0,0 +1,127 @@
+        `[URL prefix]/custom/dtr_full_ehr/mock_auth/token` endpoint to get
+        an access token to use on the request of the requests.
+
+In either case, the tests will continue from that point. Further executions of tests under
+this session will also use the selected bearer token.
+
+Note: authentication options for these tests have not been finalized and are subject to change.
+
+### Postman-based Demo
+
+If you do not have a DTR Full EHR but would like to try the tests out, you can use
+[this Postman collection](https://github.com/inferno-framework/davinci-dtr-test-kit/blob/main/config/DTR%20Full%20EHR%20Tests%20Postman%20Demo.postman_collection.json)
+to make requests against Inferno. This does not include the capability to render and complete the
+questionnaires, but does have samples of correctly and incorrectly completed QuestionnaireResponses.
+To run the tests using this approach:
+
+1. Install [postman](https://www.postman.com/downloads/).
+1. Import [this Postman collection](https://github.com/inferno-framework/davinci-dtr-test-kit/blob/main/config/DTR%20Full%20EHR%20Tests%20Postman%20Demo.postman_collection.json).
+1. Start a Da Vinci DTR Full EHR Test Suite Session.
+1. Update the postman collection configuration variables found by opening the "DTR Full EHR
+   Tests Postman Demo" collection and selecting the "Variables" tab.
+   - **base_url**: corresponds to the where the test suite session is running. Defaults to 
+   `inferno.healthit.gov`. If running in another location, see guidance on the "Overview" tab
+   of the postman collection.
+   - **access_token**: note the "Current value" (update if desired) for use later.
+1. Return to Inferno and in the test list at the left, select *2 Static Questionnaire Workflow*.
+1. Click the "Run All Tests" button in the upper right.
+1. Add the **access_token** configured in postman to the Inferno input with the same name
+1. Click the "Submit" button in Inferno.
+1. Attest that the EHR has launched its DTR workflow in Inferno by clicking the link for the **true** response.
+1. Once the next wait dialog has appeared within Inferno asking for a `$questionnaire-package` 
+   request, use postman to submit the "Questionnaire Package for Dinner (Static)" request. Confirm
+   that the response that looks similar to the "Example Working Response" in postman
+   and click the link to continue the tests.
+1. Attest to the remainder of the tests as desired to get a sense for what is involved in testing
+   with an actual EHR implementation. To see what a valid QuestionnaireResponse looks like, see
+   the "Sample QuestionnaireResponse for Dinner (Static) ..." request in postman.
+
+## Limitations
+
+The DTR IG is a complex specification and these tests currently validate conformance to only
+a subset of IG requirements. Future versions of the test suite will test further
+features. A few specific features of interest are listed below.
+
```

Comment:
Could not that while Full EHR SHALL require support of adaptive questionnaires as well (unlike payer servers), we are only covering static ones for now. It might also be good to put caveats for items we are not testing on the client side. I may have been confused by the IG here and included things that apply to only the SMART app - I am not sure how liberally the authors mean to use 'client system' or 'DTR client'.

Other items that are not covered:
- client SHALL visually flag any elements that require adjustment to meet validation rules (or is this SMART APP specific?)
- DTR applications SHALL support rendering according to the mustSupport elements 
- when the QuestionnaireResponse is valid, the DTR client SHALL indicate that to the user and allow them to mark the QuestionnaireResponse as complete
- If any of the retrieved Questionnaires have an effectivePeriod that ends prior to the current date, then the DTR client SHALL change the status of any retrieved work-in-progress QuestionnaireResponses for the expired Questionnaires to 'stopped' and notify the user that the previously recorded content has expired.
- Client SHALL support saving in-progress questionnaires (we have attestation of saving completed questionnaire)
- When a user fills in a value in a QuestionnaireResponse, the DTR client SHALL populate the information-origin extension
- DTR clients SHALL either provide a PractitionerRole in the SMART App launch of DTR or support transmitting the role by means of the [activeRole](https://build.fhir.org/ig/HL7/davinci-dtr/StructureDefinition-activeRole.html) extension within the Practitioner resource.

Reply:
Most of this will be reviewed / addressed as we dig into the requirements for the g(34/35) criteria and bring requirement coverage analysis to this test kit. I think most of this is fine to address as a part of that work and I'm comfortable with the current limitations section in the full EHR documentation.

> client SHALL visually flag any elements that require adjustment to meet validation rules (or is this SMART APP specific?)
> DTR applications SHALL support rendering according to the mustSupport elements

I think both of these are covered by the limitation section where we say that not all questionnaire capabilities are covered. I would include this as a questionnaire rendering requirement.

> when the QuestionnaireResponse is valid, the DTR client SHALL indicate that to the user and allow them to mark the QuestionnaireResponse as complete

I think this is covered by the attestations

> If any of the retrieved Questionnaires have an effectivePeriod that ends prior to the current date, then the DTR client SHALL change the status of any retrieved work-in-progress QuestionnaireResponses for the expired Questionnaires to 'stopped' and notify the user that the previously recorded content has expired.

Will consider this requirement later.

> Client SHALL support saving in-progress questionnaires (we have attestation of saving completed questionnaire)

Will consider this requirement later.

> When a user fills in a value in a QuestionnaireResponse, the DTR client SHALL populate the information-origin extension

Covered by attestations

> DTR clients SHALL either provide a PractitionerRole in the SMART App launch of DTR or support transmitting the role by means of the [activeRole](https://build.fhir.org/ig/HL7/davinci-dtr/StructureDefinition-activeRole.html) extension within the Practitioner resource.

SMART App-specific, not covered right now.